cryptoki/src/context/mod.rs

```diff
@@ -13,6 +13,14 @@ macro_rules! get_pkcs11 {
     };
 }
 
+/// Same as get_pkcs11! but does not attempt to apply '?' syntactic sugar.
+/// Suitable only if the caller can't return a Result.
+macro_rules! get_pkcs11_func {
```

Comment:
Would it work to modify `get_pkcs11` macro to use this one + the `ok_or`? Meaning to avoid code duplication :)

Reply:
Seems reasonable, I will give it a stab. Not much XP with Rust macro authoring, however. I'll have a look at it this week-end.

Reply:
here is it - easier than anticipated it seems.